libraries/shared/watcher/storage_watcher.go

```diff
@@ -90,8 +94,31 @@ func (watcher StorageWatcher) Execute() error {
 	}
 }
 
+func (watcher StorageWatcher) getMinDiffID() (int, error) {
+	var minID int
+	if watcher.DiffBlocksFromHeadOfChain != -1 {
+		mostRecentHeaderBlockNumber, getHeaderErr := watcher.HeaderRepository.GetMostRecentHeaderBlockNumber()
+		if getHeaderErr != nil {
+			return 0, getHeaderErr
+		}
+		blockNumber := mostRecentHeaderBlockNumber - watcher.DiffBlocksFromHeadOfChain
+		diffID, getDiffErr := watcher.StorageDiffRepository.GetFirstDiffIDForBlockHeight(blockNumber)
+		if getDiffErr != nil {
+			return 0, getDiffErr
+		}
+
+		minID = int(diffID - 1)
+	}
+
+	return minID, nil
```

Comment:
So if `DiffBlocksFromHeadOfChain` is -1 you'll return for minID. Is that deliberate?

Reply:
Yes, the idea is that if `DidffBlocksFromHeadOfChain` is -1 then we'll want the `minID` to be 0. Golang sets the starting value of an int to 0 if you don't define it (line 98), but perhaps it would be clearer to be explicit and set `var minID= 0` instead.

Reply:
I think it's the "-1" that's bothering me - it's a magic number that the `execute.go` file has to know about to start from the beginning.

Honestly it's not super important, just mildly confusing.

Reply:
That's super fair w/r/t the magic number! I've added a comment explaining a bit more, and may also pull the -1 out into a variable/constant to give a bit more context.